src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature

```diff
@@ -0,0 +1,194 @@
+    Then Enter input plugin property: "dataset" with value: "dataset"
+    Then Enter input plugin property: "table" with value: "bqTargetTable"
+    And Select radio button plugin property: "operation" with value: "update"
+    Then Click plugin property: "updateTableSchema"
+    Then Enter the Table Key property for performing update and upsert operation with value: "TableKey"
+    Then Validate "BigQuery" plugin properties
+    And Close the Plugin Properties page
+    Then Save the pipeline
+    Then Preview and run the pipeline
+    Then Wait till pipeline preview is in running state
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "succeeded"
+    Then Close the pipeline logs
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Close the pipeline logs
+    Then Verify the pipeline status is "Succeeded"
+
+  @BQ_INSERT_SOURCE_TEST @BQ_SINK_TEST
+  Scenario:Validate successful records transfer from BigQuery to BigQuery with Advanced operations Upsert
+    Given Open Datafusion Project to configure pipeline
+    When Expand Plugin group in the LHS plugins list: "Source"
+    When Select plugin: "BigQuery" from the plugins list as: "Source"
+    When Expand Plugin group in the LHS plugins list: "Sink"
+    When Select plugin: "BigQuery" from the plugins list as: "Sink"
+    Then Connect plugins: "BigQuery" and "BigQuery2" to establish connection
+    Then Navigate to the properties page of plugin: "BigQuery"
+    And Replace input plugin property: "project" with value: "projectId"
+    Then Override Service account details if set in environment variables
+    And Replace input plugin property: "datasetProject" with value: "datasetprojectId"
+    And Replace input plugin property: "referenceName" with value: "reference"
+    And Replace input plugin property: "dataset" with value: "dataset"
+    And Replace input plugin property: "table" with value: "bqSourceTable"
+    Then Click on the Get Schema button
+    Then Validate "BigQuery" plugin properties
+    And Close the Plugin Properties page
+    Then Navigate to the properties page of plugin: "BigQuery2"
+    Then Replace input plugin property: "project" with value: "projectId"
+    Then Override Service account details if set in environment variables
+    Then Enter input plugin property: "datasetProject" with value: "projectId"
+    Then Enter input plugin property: "referenceName" with value: "BQReferenceName"
+    Then Enter input plugin property: "dataset" with value: "dataset"
+    Then Enter input plugin property: "table" with value: "bqTargetTable"
+    And Select radio button plugin property: "operation" with value: "upsert"
+    Then Click plugin property: "updateTableSchema"
+    Then Enter the Table Key property for performing update and upsert operation with value: "TableKey"
```

Comment:
Replace this step with a generic step definiton like: https://github.com/cdapio/cdap-e2e-tests/blob/4ec16f7af29ffcfb5e44cf549ec7a919cf44d2fa/src/main/java/stepsdesign/PipelineSteps.java#L457

Reply:
Adding a generic step according to these fields (tableKey, deDupeBy and clusteringOrder) as a locator is different from the one which we used in Cdap e2e framework step.

Reply:
I know it is different, my comment was to add a similar generic locator in cdap-e2e-framework and use it here.

Reply:
Added.